elevated-access (#1600)
collective/icalendar: docs/contribute/join-the-icalendar-team.rst

=======================
Join the icalendar team
=======================

This guide describes how to join the icalendar team.

Purpose
-------

The icalendar team reviews pull requests and issues for quality, makes releases to PyPI, and protects and promotes the icalendar community.
To familiarize yourself with the icalendar community, read :doc:`index`.

> To encourage collaboration and community, the icalendar team is actively seeking contributors who are interested in elevated access to icalendar to help review and triage issues and pull requests.

Reviewer: I would say we are seeking people who are interested in doing the work. I do not really want those who seek the access as primary motivation.

Author: Done in 94d6c2ce7c543de03bed82a0ab766f2889684d30

A good candidate for elevated access demonstrates the following qualifications:

-   Friendliness, humanity, and values diversity and inclusion.
-   Frequently engage with other contributors.
-   Frequently triage and review pull requests and issues.

If you're interested in elevated access, you may:

-   Create a new `Discussion topic <https://github.com/collective/icalendar/discussions/categories/collaborator-nominations>`_ under the Collaborator nominations category.
-   Post a message to the `Matrix icalendar chat <https://matrix.to/#/#pycal:chat.pycal.org>`_.

> Additionally, the icalendar team may contact you to encourage you to apply.

Reviewer: ```suggestion
Additionally, the icalendar team may contact you to encourage you to join.
```

Author: Same.

> The icalendar team recognizes that many individuals might not feel qualified to apply. Please rest assured that the worst that can happen is that the team will say "not yet, but here's a path you can follow to become qualified." The icalendar team wants to lift people up.

Reviewer: Can we say "we" instead of icalendar team?

Author: The Microsoft Style Guide recommends avoiding "we" to focus the subject on the reader, not the product. However, that's a different context.

In this context, "We" may have a negative connotation of exclusivity, or "us, not you, the reader". If you're not a member of "we", then that can be a turn off. At other times, it may include the reader, making it unclear who "we" consists of. I could explicitly state "'We' in this document refers to the icalendar team." but I prefer to explicitly state the subject. It also reinforces that we're a team.

> The icalendar team values diversity, equity, and inclusion. The icalendar team encourages women, people of color, LGBTQ+ individuals, people with disabilities, neuro-divergent people, members of ethnic minorities, residents of countries outside western culture, people of all ages, and people whose primary language is not English to apply. Members of the icalendar team are committed to building a team that is inclusive of a variety of backgrounds and perspectives.

Reviewer: ```suggestion
The icalendar team values diversity, equity, and inclusion. The icalendar team encourages women, people of color, LGBTQ+ individuals, people with disabilities, neuro-divergent people, members of ethnic minorities, residents of countries outside western culture, people of all ages, and people whose primary language is not English to join. Members of the icalendar team are committed to building a team that is inclusive of a variety of backgrounds and perspectives.
```

Author: "join" is not correct here. When they accept the invitation, then they join. "apply" is correct.

Reviewer: If join >= apply + accept, I would like to also encourage them to join. Especially, since join for me means to put the work in.
Join also ties it to the title.

Author: join = apply + review + invite + accept invitation. It's a process. If they're not invited, then they can't join yet, but we give them advice for how to qualify.

Here's another way to look at it. You announce you will throw a party on social media. An AI bot reads the invitation, takes human form, shows up and joins the party, then proceeds to squat over the punch bowl and drops a load 💩. This is why we have a process: to ensure that we don't allow poopers to join the party.

Reviewer: I would like to distinct these:

- There is a process of how people joined. We can describe the past.
- There are requirements for triage. That is different from a process of joining.
- The described process is different from the past process. The past process is not described.

I think that any process applied that fulfills the minimal requirements is good enough. It is good to describe one process that works as a guideline, try, evaluate, adapt.

So, I have no issue with the requirements. I will look at the process. I will have an issue with sacrificing talent for bürocracy and like to err on the side of chaos and trust as that expands consciousness and invites creativity.

Let's have Sashank proofread it and also ask a few collaborarors with write access if they would have gone through the process and it would have worked for them.

Author: > * There is a process of how people joined. We can describe the past.

I don't think that's helpful in this context. That would be explanation, in Diataxis, of what we did and how we got here. This is a how-to.

> * There are requirements for triage. That is different from a process of joining.

I think that if people don't triage or review issues and PRs, then they shouldn't apply to join the team. Triage is a requirement to become eligible.

> * The described process is different from the past process. The past process is not described.

See my first response. And I'd argue that there was no process to join the team because there was no icalendar team.

> So, I have no issue with the requirements. I will look at the process. I will have an issue with sacrificing talent for bürocracy and like to err on the side of chaos and trust as that expands consciousness and invites creativity.

We must not allow punch bowl party poopers. Now _that_ would be chaos. I think the proposed process is reasonable because it prevents both unwanted chaos and unwarranted access. Granting Triage access shows trust and commitment with one another, while preventing the chaos from a potentially compromised account. This protects the team, too, keeping the space safe. I feel it is far more important to protect what works than risk it falling apart.

As an aside, in a subsequent PR, I'd revise this to include Write and Maintainer access, but we need to discuss those separately.

-   Apply and dismiss labels.
-   Close, reopen, and assign all issues and pull requests.
-   Apply milestones.
-   Mark duplicate issues and pull requests.
-   Request pull request reviews.
-   Hide anyone's comments.
-   Move a discussion to a different category.
-   Lock and unlock discussions.
-   Individually convert issues to discussions.
-   Delete a discussion.

The primary criteria that the icalendar team uses to consider whether to grant elevated access include, but are not limited to:

-   Previous contributions to icalendar and related projects, especially pull request reviews and issue triage
-   Previous contributions to other projects
-   Demonstrated understanding of the RFCs and code base without AI assistance
-   Two-factor authentication must be enabled on their GitHub account
-   Strive to uphold the spirit of the Python Calendaring Ecosystem's `Code of Conduct <https://pycal.org/code-of-conduct/>`_
-   Join the `#icalendar Matrix chat <https://matrix.to/#/#pycal:chat.pycal.org>`_
-   Join a video call in Matrix

The icalendar team will review your application, and collect relevant information for evaluation.

> If the icalendar team determines that you qualify for elevated access, then you'll receive an invitation through an email notification from GitHub to accept.

Reviewer: We have the maintenance page - we do not need to be a faceless team but people can know who is the team - should that be restructured then?

A few friendly faces with roles.
Hi, we are the team behind icalendar.
Join us... Here is how...

Then three tabs

| Review | Triage | Maintain |
| --- | --- | --- |
| What we really need is review work. We get a lot of contributions and this helps scale up. When you review, you show that you know how to be a team player, not just a coder. | Issues need to be checked and PRs merged. Once you showed that you can review, we can trust you with your judgement on issues and PRs. You know how to cooperate. | It is always good for the project to have a few maintainers.... ???? |

Author: Maintenance is written primarily for people who are interested in maintaining icalendar. This page's purpose is to describe how to join the icalendar team. Let's not mix their purposes.

I could add a section about who are members of the icalendar team, which coincidentally are those people listed on Maintenance.

For the tabs, Review can be done by anyone with a GitHub account via Read access. Maintainers is outside the scope of this PR, and needs a separate discussion about who may become one. I focused on Triage and getting people to join the icalendar team. In another PR, we could amend this page with how to be promoted to Maintainer, once we determine those qualifications.

Author: Added a Members section in https://github.com/collective/icalendar/pull/1600/commits/b7dd62a1295839241fb9202dfd3e0a9be950d676

Author: I should remark that GitHub has shitty permission management for FOSS organizations. Access may be granted on the entire organization and all its repos, but not a singular repo. Instead we must use direct individual access.

We can also create an organization team, such as `icalendar-triage` or `icalendar-maintainers`, and add members to each team, but that won't change their individual access. It's only practical use is to tag a team or request a review from a team, such as @icalendar-maintainers`, which would send a notification to its members.

Reviewer: Do we have a section somewhere that describes what people can do before they are granted access? E.g review, comment and check issues?

Author: No, but we don't need to, as GitHub already publishes the Read role and its permissions under "repository roles for an organization", mentioned above. I don't want to duplicate that documentation. It could change without our team knowing.

=======================
Join the icalendar team
=======================

This guide describes how to join the icalendar team.

Purpose
-------

The icalendar team reviews pull requests and issues for quality, makes releases to PyPI, and protects and promotes the icalendar community.
To familiarize yourself with the icalendar community, read :doc:`index`.

To encourage collaboration and community, the icalendar team is actively seeking contributors who are interested in helping to review and triage issues and pull requests.

Good candidates for the team demonstrate the following qualifications:

-   Friendliness, humanity, and value diversity and inclusion.
-   Frequently engage with other contributors.

> -   Frequently triage and review pull requests and issues.

Reviewer 2: I think line 18 reads as a noun phrase but 19 and 20 are verb phrases? What do you think about this suggestion for consistency?

```suggestion
-   Be friendly, humane, and value diversity and inclusion.
-   Frequently engage with other contributors.
-   Frequently triage and review pull requests and issues.
```

Author: Good catch. Added to batch.

> -   Frequently triage and review pull requests and issues.

Reviewer: Narrative person mismatch 18 - 3rd person, 19,20 unconjugated

Author: Fixed in https://github.com/collective/icalendar/pull/1600/commits/9bb524c89c3bf240dd80ffc51437ddeb2d36662c

Members
-------

> The members of the icalendar team include its maintainers and documentation lead.

Reviewer: ```suggestion
The members of the icalendar team are listed below in their respective roles that they fulfill in this project.
```

Author: Done in https://github.com/collective/icalendar/pull/1600/commits/6c8b8715bbdabea35f4d652f631978a7656ee137

Maintainers
```````````

.. include:: ../_include/maintainers.inc

Documentation lead
``````````````````

`@stevepiercy <https://github.com/stevepiercy>`_

Triage
``````

Apply to add your name here.

> 

Reviewer: I would add a section here for contributors to contrast.

```suggestion
Contributors

Contributors are those people who contributed code to icalendar. In contrast to team members, they do not review PRs and are only active for a short amount of time.
They are listed here: Link
```

Or this can go into a related note at the bottom.

---

In a future PR or this PR, we can list past maintainers and such. 

What do you think?

Author: This page is focused on people who want to become part of the team, not a Contributor role, which is already documented in the Contribute guide.

Apply
-----

> If you're interested in applying to join the team, you may:

Reviewer: ```suggestion
If you're interested in applying to join the team or if you have found someone who would like to join, you may:
```
How is that?

Author: No. If someone is truly interested, then they should speak for themselves, not through a proxy.

-   Create a new `Discussion topic <https://github.com/collective/icalendar/discussions/categories/collaborator-nominations>`_ under the Collaborator nominations category.
-   Post a message to the `Matrix icalendar chat <https://matrix.to/#/#pycal:chat.pycal.org>`_.

Additionally, the icalendar team may contact you to encourage you to apply.

The icalendar team recognizes that many individuals might not feel qualified to apply. Please rest assured that the worst that can happen is that the team will say "not yet, but here's a path you can follow to become qualified." The icalendar team wants to lift people up.

The icalendar team values diversity, equity, and inclusion. The icalendar team encourages women, people of color, LGBTQ+ individuals, people with disabilities, neuro-divergent people, members of ethnic minorities, residents of countries outside western culture, people of all ages, and people whose primary language is not English to apply. Members of the icalendar team are committed to building a team that is inclusive of a variety of backgrounds and perspectives.

Elevated access roles
---------------------

> The icalendar team may grant elevated levels of access—what GitHub calls "roles"—to the icalendar repository for contributors based on qualifying criteria.

Reviewer: Is the spacing right? I was confused to read the word accesswhat, just dashed.

Author: Yes. It's an em-dash, but the monospace font makes it appear too much like a hyphen. When rendered to text, you can see the difference

- hyphen: -
- em-dash: —

The Triage role has the following additional permissions:

-   Apply and dismiss labels.
-   Close, reopen, and assign all issues and pull requests.
-   Apply milestones.
-   Mark duplicate issues and pull requests.
-   Request pull request reviews.
-   Hide anyone's comments.
-   Move a discussion to a different category.
-   Lock and unlock discussions.
-   Individually convert issues to discussions.
-   Delete a discussion.

The primary criteria that the icalendar team uses to consider whether to grant elevated access include, but are not limited to:

-   Previous contributions to icalendar and related projects, especially pull request reviews and issue triage
-   Previous contributions to other projects
-   Demonstrated understanding of the RFCs and code base without AI assistance
-   Two-factor authentication must be enabled on their GitHub account
-   Strive to uphold the spirit of the Python Calendaring Ecosystem's `Code of Conduct <https://pycal.org/code-of-conduct/>`_
-   Join the `#icalendar Matrix chat <https://matrix.to/#/#pycal:chat.pycal.org>`_

> -   Join a video call in Matrix

Reviewer: ```suggestion
-   Join a video call to get to know each other
```

The icalendar team will review your application, and collect relevant information for evaluation.

If the icalendar team determines that you qualify for elevated access, then you'll receive an invitation through an email notification from GitHub to accept.

> If the team determines you're not yet qualified for elevated access, then you'll receive information about next steps to qualify.

Reviewer 2: The doc focuses on Triage access but a reader has no way of knowing there's a Collaborator level above it.

 `maintenance.rst` already describes that path. 

Maybe one sentence at the end pointing to it would make the progression clear rather than leaving Triage looking like a dead end?
